### PR TITLE
A rd 5221 alb triggered by support

### PR DESCRIPTION
**Pull request by sapirlumigo**

### CODE REVIEW

File: src/lumigo_tracer/event/event_trigger.py [hunk 1]
```diff
@@ -77,7 +79,20 @@ def _is_supported_http_method(event: dict):
...
+def _is_load_balancer_method(event: dict):
+    return (
+        "httpMethod" in event  # noqa
+        and "headers" in event  # noqa
+        and event["headers"].get("host")  # noqa
+        and "requestContext" in event  # noqa
+        and (  # noqa
+            event.get("requestContext", {}).get("elb") is not None  # noqa
+            or event.get("requestContext", {}).get("alb") is not None  # noqa
+        )  # noqa
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:94]:
```suggestion
        ) 
```
I think

File: src/lumigo_tracer/event/event_trigger.py [hunk 2]
```diff
@@ -77,7 +79,20 @@ def _is_supported_http_method(event: dict):
...
+
+
+def _is_load_balancer_method(event: dict):
+    return (
+        "httpMethod" in event  # noqa
+        and "headers" in event  # noqa
+        and event["headers"].get("host")  # noqa
+        and "requestContext" in event  # noqa
+        and (  # noqa
+            event.get("requestContext", {}).get("elb") is not None  # noqa
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:92]:
```suggestion
            event.get("requestContext", {}).get("elb") is not None 
```
I think

File: src/lumigo_tracer/event/event_trigger.py [hunk 3]
```diff
@@ -87,6 +102,20 @@ def _parse_http_method(event: dict):
     return _parse_http_method_v1(event)
 
 
+def _parse_load_balancer_method(event: dict):
+    result = {
+        "triggeredBy": "load_balancer",
+        "httpMethod": event.get("httpMethod", ""),
+        "resource": event.get("resource", ""),
+        "messageId": event.get("requestContext", {}).get("requestId", ""),
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:110]: I don't think that this ever exists..
https://docs.aws.amazon.com/lambda/latest/dg/services-alb.html

File: src/lumigo_tracer/event/event_trigger.py [hunk 4]
```diff
@@ -87,6 +102,20 @@ def _parse_http_method(event: dict):
     return _parse_http_method_v1(event)
 
 
+def _parse_load_balancer_method(event: dict):
+    result = {
+        "triggeredBy": "load_balancer",
+        "httpMethod": event.get("httpMethod", ""),
+        "resource": event.get("resource", ""),
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:109]: never exists

File: src/lumigo_tracer/event/event_trigger.py [hunk 5]
```diff
@@ -87,6 +102,20 @@ def _parse_http_method(event: dict):
...
+        "messageId": event.get("requestContext", {}).get("requestId", ""),
+    }
+    if isinstance(event.get("headers"), dict):
+        result["api"] = event["headers"].get("host")
+    if isinstance(event.get("requestContext"), dict):
+        result["stage"] = event["requestContext"].get("stage")
```
Comment (saartochner) [inline src/lumigo_tracer/event/event_trigger.py:115]: can you try to create an elb/alb with stage?
I don't know if this is possible...

File: src/test/unit/event/test_event_trigger.py [hunk 6]
```diff
@@ -302,7 +302,82 @@
...
+                },
+                "isBase64Encoded": False,
+                "body": "request_body",
+            },
+            {
+                "api": "lambda-111-us-east-2.elb.amazonaws.com",
+                "httpMethod": "GET",
+                "messageId": "",
+                "resource": "",
+                "stage": None,
```
Comment (saartochner) [inline src/test/unit/event/test_event_trigger.py:378]: this is a good hint that these fields are unnecessary 🤔

File: src/lumigo_tracer/event/event_trigger.py (hunk 2 above)
Comment (sapirlumigo) [reply src/lumigo_tracer/event/event_trigger.py:92]: src/lumigo_tracer/event/event_trigger.py:93:13: W503 line break before binary operator

File: src/lumigo_tracer/event/event_trigger.py (hunk 1 above)
Comment (sapirlumigo) [reply src/lumigo_tracer/event/event_trigger.py:94]: src/lumigo_tracer/event/event_trigger.py:93:13: W503 line break before binary operator

**Review by saartochner — CHANGES_REQUESTED** [review]
GJ!

**Review by saartochner — APPROVED** [review]

**Pull request merged**